src/main/java/io/fabric8/maven/docker/assembly/DockerAssemblyManager.java

```diff
@@ -130,7 +130,9 @@ public File createDockerTarArchive(String imageName, final MojoParameters params
                 archiveCustomizers.add(new ArchiverCustomizer() {
                     @Override
                     public TarArchiver customize(TarArchiver archiver) throws IOException {
-                        DefaultFileSet fileSet = DefaultFileSet.fileSet(dockerFile.getParentFile());
+                        DefaultFileSet fileSet = DefaultFileSet.fileSet(buildConfig.getContextDir() != null
```

Comment:
According to your change below, `buildConfig.getContextDir()` can never be null.